### DISCUSSION TITLE
Requests are broken after request with body

### DISCUSSION THREAD

**Discussion by sgatu** [mistweaverco/kulala.nvim#242] [Q&A] [answered]
Not sure why but after the "Create template" request the rest are broken.
When I try to execute the "Get template" I get a "No request found".
When I execute the "Get all templates" it executes the "Get email".
When I execute the "Get email" I get again an "No request found". 

I imagine something of this file breaks the parsing. Not sure if it's my fault or something it's wrong with the library. 

[api.txt](https://github.com/user-attachments/files/17007023/api.txt)

**Comment by Grueslayer** [comment]
Hi,

please add a ### line after the variables set, before the first request, that could fix it

personally I would add an empty line after ### lines and an additional after comment and before request line.

**Reply by sgatu** [reply]
Still the same problem, I'm pretty sure is the "Create a template" request because even if it runs ok, the checkmark appears above the "Create a template" comment and not next to the url like the rest above it. 
[api.txt](https://github.com/user-attachments/files/17007076/api.txt)

What I mean by the checkmark in wrong position:
![Snapshot_2024-09-15_20-09-46](https://github.com/user-attachments/assets/eed09c04-5c42-4c5a-b497-7f31316ea6b2)

**Reply by Grueslayer** [reply]
When I remove the double empty lines, all requests seems to match the request lines.

(in this file I also add empty lines after comments, replaced tabs with spaces in the json payload)

[api.txt](https://github.com/user-attachments/files/17007124/api.txt)

**Comment by gorillamoe** [comment, marked as answer]
This here should work:

```http
@host = http://localhost:3000
@authorization = Bearer xxxxxxxxxxxxxxxxxxxxxxxxx
@domain_id = 1834991192461910016
@template_id = 1835007224123596800
@email_id = 1832899994003808256

# Get domains endpoint

GET {{host}}/domain
Authorization: {{authorization}}

###

# Get single domain

GET {{host}}/domain/{{domain_id}}
Authorization: {{authorization}}

###

# Create new domain

POST {{host}}/domain
Authorization: {{authorization}}
Content-Type: application/json

{
	"name": "domain.tld",
	"region": "eu-west-1"
}

###

# Delete domain

DELETE {{host}}/domain/{{domain_id}}?full=true
Authorization: {{authorization}}

###

# Create a template

POST {{host}}/template
Authorization: {{authorization}}
Content-Type: application/json

{
	"html": "<div>Hello [[FIRST_NAME]]</div>",
	"text": "Hello [[FIRST_NAME]]",
	"subject": "Salutations from [[COMPANY_NAME]]"
}

###

# Get template

GET {{host}}/template/{{template_id}}
Authorization: {{authorization}}

###

# Get all templates

GET {{host}}/template
Authorization: {{authorization}}

###

# Get email

GET {{host}}/email/{{email_id}}
Authorization: {{authorization}}
```

Removed additional new lines.

Could probaly also have been fixed by using [kulala-fmt](https://github.com/mistweaverco/kulala-fmt)

**Reply by sgatu** [reply]
Thank you very much. It now works, first time I'm working with .http files and I'm always doubting how many empty lines, how many comments I can write ... etc. I'll get used to it, but the kulala-fmt will be of great help.

**Reply by Grueslayer** [reply]
@gorillamoe Du bist ja eh schneller...
